oem/ibm/libpldmresponder/file_io_type_smsmenu.cpp

```diff
@@ -0,0 +1,183 @@
+#include "file_io_type_smsmenu.hpp"
+
+#include "libpldm/base.h"
+#include "libpldm/file_io.h"
+
+#include "pfw-sms-utils/pfw_sms_menu.hpp"
+
+#include <stdint.h>
+
+#include <iostream>
+
+namespace pldm
+{
+using namespace pldm::responder::utils;
+using namespace ibm_pfw_sms;
+
+namespace responder
+{
+
+int SmsMenuHandler::write(const char* buffer, uint32_t /*offset*/,
+                          uint32_t& length,
+                          oem_platform::Handler* /*oemPlatformHandler*/,
+                          uint8_t& fileStatus, uint32_t& fileMetaData1,
+                          uint32_t& fileMetaData2, uint32_t& fileMetaData3,
+                          uint32_t& fileMetaData4)
+{
+    int rc = PLDM_SUCCESS;
+
+    if (buffer != nullptr)
```

Comment:
we can directly check if (buffer != nullptr) and return based on that. can enhance this.